mongoengine/queryset/base.py

```diff
@@ -355,8 +355,8 @@ def insert(
         except pymongo.errors.BulkWriteError as err:
             # inserting documents that already have an _id field will
             # give huge performance debt or raise
-            message = u"Document must not have _id value before bulk write (%s)"
-            raise NotUniqueError(message % six.text_type(err))
+            message = u"Bulk write error: (%s)"
```

Comment:
Is it still correct to use `NotUniqueError` or should we add a `BulkWriteError` ? it would be a breaking change though, I guess it depends if there are other cases that triggers pymongo's BulkWriteError than simply the Non-Uniqueness of documents. How exactly is this exception triggered?

Reply:
Agree. :) The error was index key too large. I will add BulkWriteError